src/gsNurbs/gsCurveCurveIntersection.h

```diff
@@ -0,0 +1,360 @@
+/** @file gsCurveCurveIntersection.h
+
+    @brief Implementation of B Spline Curve/Curve intersection
+
+    This file is part of the G+Smo library.
+
+    This Source Code Form is subject to the terms of the Mozilla Public
+    License, v. 2.0. If a copy of the MPL was not distributed with this
+    file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+    Author(s): Ye Ji
+*/
+
+#pragma once
+
+#include <gsCore/gsLinearAlgebra.h>
+
+namespace gismo{
+
+namespace internal {
+
+#define EPSILON_CCI ( 100*std::numeric_limits<T>::epsilon() )
+
+template<class T=real_t>
+struct gsPoint2d{
+  gsPoint2d(T xCoord, T yCoord) {
+    pt.x() = xCoord;
+    pt.y() = yCoord;
+  }
+
+  inline T x() const { return pt.x();}
+  inline T y() const { return pt.y();}
+
+  gsPoint<2,T> pt;
+};
+
+// To find orientation of ordered triplet (p, q, r).
+// The function returns following values
+// 0 --> p, q and r are collinear
+// 1 --> Clockwise
+// 2 --> Counterclockwise
+template<class T>
+int orientationxx(const gsPoint2d<T> &p, const gsPoint2d<T> &q, const gsPoint2d<T> &r) {
+  // See https://www.geeksforgeeks.org/orientation-3-ordered-points/
+  // for details of below formula.
+  double val = (q.y() - p.y()) * (r.x() - q.x()) - (q.x() - p.x()) * (r.y() - q.y());
+
+  if (val == 0) return 0;  // collinear
+
+  return (val > 0) ? 1 : 2; // clock or counterclock wise
+}
+
+// Given three collinear points p, q, r, the function checks if
+// point q lies on the line segment 'pr'
+template<class T>
+bool onSegment(const gsPoint2d<T> &p, const gsPoint2d<T> &q, const gsPoint2d<T> &r) {
+  if (q.x() <= std::max(p.x(), r.x()) && q.x() >= std::min(p.x(), r.x()) &&
```

Comment:
please use math::max instead of std::max
and math::min instead of std::min
everywhere

Reply:
Done!